pkg/api/aim/experiments.go

```diff
@@ -327,3 +335,77 @@ func DeleteExperiment(c *fiber.Ctx) error {
 		"status": "OK",
 	})
 }
+
+func UpdateExperiment(c *fiber.Ctx) error {
+	ns, err := namespace.GetNamespaceFromContext(c.Context())
+	if err != nil {
+		return api.NewInternalError("error getting namespace from context")
+	}
+	log.Debugf("updateExperiment namespace: %s", ns.Code)
+
+	params := struct {
+		ID string `params:"id"`
+	}{}
+	if err = c.ParamsParser(&params); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	var updateRequest request.UpdateExperimentRequest
+	if err = c.BodyParser(&updateRequest); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	id, err := strconv.ParseInt(params.ID, 10, 32)
+	if err != nil {
+		return api.NewBadRequestError("Unable to parse experiment id '%s': %s", params.ID, err)
+	}
+	experimentRepository := repositories.NewExperimentRepository(database.DB)
+	tagRepository := repositories.NewTagRepository(database.DB)
+	experiment, err := experimentRepository.GetByNamespaceIDAndExperimentID(c.Context(), ns.ID, int32(id))
+	if err != nil {
+		return fiber.NewError(
+			fiber.StatusInternalServerError, fmt.Sprintf("unable to find experiment '%s': %s", params.ID, err),
+		)
+	}
+	if experiment == nil {
+		return fiber.NewError(fiber.StatusNotFound, fmt.Sprintf("unable to find experiment '%s'", params.ID))
+	}
+	if updateRequest.Archived != nil {
+		if *updateRequest.Archived {
+			experiment.LifecycleStage = models.LifecycleStageDeleted
+		} else {
+			experiment.LifecycleStage = models.LifecycleStageActive
+		}
+	}
+
+	if updateRequest.Name != nil {
+		experiment.Name = *updateRequest.Name
+	}
+
+	if updateRequest.Archived != nil || updateRequest.Name != nil {
+		if err := database.DB.Transaction(func(tx *gorm.DB) error {
+			if err := experimentRepository.UpdateWithTransaction(c.Context(), tx, experiment); err != nil {
+				return err
+			}
+			return nil
+		}); err != nil {
+			return fiber.NewError(fiber.StatusInternalServerError,
+				fmt.Sprintf("unable to update experiment %q: %s", params.ID, err))
+		}
+	}
+	if updateRequest.Description != nil {
+		description := models.ExperimentTag{
+			Key:          "mlflow.note.content",
```

Comment:
i'm wondering if this should be a first-class field on Experiment table instead of using a magic tag. We could then translate experiment.description -> experiment_tag on the mlflow side where needed. That way the mlflow implementation (magic tag) would be isolated on the mlflow side of the house.

Reply:
this is a great idea for a future improvement!
for now I would keep using the magic tag so we can get this gap bridged quickly